src/Gui/ViewProviderPyImp.cpp

```diff
@@ -392,20 +413,21 @@ PyObject* ViewProviderPy::partialRender(PyObject* args)
         }
         values.resize(nSize);
         for (Py_ssize_t i = 0; i < nSize; ++i) {
-            if(value) item = PySequence_GetItem(value, i);
+            if(value)
+                item = PySequence_GetItem(value, i);
             if (PyUnicode_Check(item)) {
                 values[i] = PyUnicode_AsUTF8(item);
             }
             else {
-                std::string error = std::string("type must be str or unicode");
+                std::string error = std::string("type must be str");
                 error += " not, ";
                 error += item->ob_type->tp_name;
                 throw Base::TypeError(error);
             }
         }
     }
 
-    Py::Int ret(getViewProviderPtr()->partialRender(values,PyObject_IsTrue(clear)));
+    Py::Int ret(getViewProviderPtr()->partialRender(values,clear));

```

Comment:
@marioalexis84 
This change doesn't look correct to me because `clear` is a pointer to a `PyObject` which will never become null. Passing the non-null pointer to a function that accepts a `bool` the value will always be true.
Because `PyObject_IsTrue` returns an `int` some compilers may raise a warning. So, the best way would be to write `PyObject_IsTrue(clear) ? true : false`.

Reply:
>Passing the non-null pointer to a function that accepts a bool the value will always be true.

Yes. Thanks for the fix.
>Because PyObject_IsTrue returns an int some compilers may raise a warning.

There are several places where PyObject_IsTrue is used without the boolean conversion. I will prepare the changes for this.